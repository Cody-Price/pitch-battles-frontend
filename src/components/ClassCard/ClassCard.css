.class-card,
.new-class-card {
  width: 250px;
  border-radius: 30px;
  box-shadow: 3px 3px 3px rgba(0, 0, 0, 0.2);
  margin: 1rem auto;
  padding-bottom: 1rem;
  height: 150px;
  font-family: "Open Sans", sans-serif;
}

.class-card-header,
.new-class-card-label {
  background: rgba(41, 67, 199, 1);
  /* background: -moz-linear-gradient(
    left,
    rgba(41, 67, 199, 1) 0%,
    rgba(101, 171, 236, 1) 100%
  );
  background: -webkit-gradient(
    left top,
    right top,
    color-stop(0%, rgba(41, 67, 199, 1)),
    color-stop(100%, rgba(101, 171, 236, 1))
  );
  background: -webkit-linear-gradient(
    left,
    rgba(41, 67, 199, 1) 0%,
    rgba(101, 171, 236, 1) 100%
  );
  background: -o-linear-gradient(
    left,
    rgba(41, 67, 199, 1) 0%,
    rgba(101, 171, 236, 1) 100%
  );
  background: -ms-linear-gradient(
    left,
    rgba(41, 67, 199, 1) 0%,
    rgba(101, 171, 236, 1) 100%
  );
  background: linear-gradient(
    to right,
    rgba(41, 67, 199, 1) 0%,
    rgba(101, 171, 236, 1) 100%
  );
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#2943c7', endColorstr='#65abec', GradientType=1 ); */
  border-top-left-radius: 30px;
  border-top-right-radius: 30px;
  color: #f2f2f2;
}

.new-class-card input {
  border-radius: 10px;
  padding-left: 0.3rem;
}

.new-class-card .new-class-button {
  display: block;
  margin: 1rem auto;
}

.class-card-name {
  margin: 0;
  padding: 1rem;
}

.new-class-card-label {
  padding: 1rem;
  margin: 0 0 1rem;
}

.class-card-key {
  font-family: "Oswald", sans-serif;
}
