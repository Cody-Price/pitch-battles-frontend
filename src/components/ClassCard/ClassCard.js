import React, { Component } from 'react';
import './ClassCard.css';

class ClassCard extends Component {
	constructor() {
		super();
		this.state = {

		}
	}

	render() {
		return (
			<article className='class-card'>
				<header className='class-card-header'>
					<h3 className='class-card-name'>{this.props.klass.name}</h3>
				</header>
				<p className='class-card-key'>{this.props.klass.key}</p>
				<button className='class-select-button' onClick={() => this.props.classSelect(this.props.klass.id)}>select</button>
			</article>
		)
	}
}

export default ClassCard;