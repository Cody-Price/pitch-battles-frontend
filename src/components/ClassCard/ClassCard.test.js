import React from "react";
import ReactDOM from "react-dom";
import ClassCard from "./ClassCard";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("ClassCard", () => {
  let mockGenerateClassCards;
  const mockData = {
    attributes: {
      name: "5th Grade Band",
      key: "keeeeeey"
    },
    id: 1
  };
  let wrapper;
  let mockClassSelect;

  beforeEach(() => {
    mockGenerateClassCards = jest.fn();
    mockClassSelect = jest.fn();
    wrapper = shallow(
      <ClassCard
        data={mockData}
        classSelect={mockClassSelect}
        generateClassCards={mockGenerateClassCards}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call classSelect on click with the correct params", () => {
    wrapper.find(".class-select-button").simulate("click");

    expect(mockClassSelect).toBeCalledWith(1);
  });

  describe("confirmDelete", () => {
    it("should call confirmDelete on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "confirmDelete");
      wrapper.instance().forceUpdate();

      wrapper.find(".fa-trash-alt").simulate("click");

      expect(spy).toHaveBeenCalled();
    });

    it("should call setState", () => {
      wrapper.instance().confirmDelete();

      expect(wrapper.state().showConfirmDelete).toEqual(true);
    });
  });

  describe("deleteClass", () => {
    it("should call setState", async () => {
      wrapper.setState({
        deleteError: true
      });
      await wrapper.instance().deleteClass();

      expect(wrapper.state().deleteError).toEqual(false);
    });

    it("should call generateClassCards", async () => {
      await wrapper.instance().deleteClass();

      expect(mockGenerateClassCards).toHaveBeenCalled();
    });

    it("should call deleteClass on click", async () => {
      const spy = jest.spyOn(wrapper.instance(), "deleteClass");

      wrapper.instance().forceUpdate();

      await wrapper.find(".class-delete-btn").simulate("click");

      expect(spy).toHaveBeenCalled();
    });
  });

  describe("doNotDelete", () => {
    it("should setstate", () => {
      wrapper.setState({
        showConfirmDelete: true
      });

      wrapper.instance().doNotDelete();

      expect(wrapper.state().showConfirmDelete).toEqual(false);
    });

    it("should call doNotDelete on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "doNotDelete");

      wrapper.instance().forceUpdate();

      wrapper.find(".class-do-not-delete-btn").simulate("click");

      expect(spy).toHaveBeenCalled();
    });
  });
});
