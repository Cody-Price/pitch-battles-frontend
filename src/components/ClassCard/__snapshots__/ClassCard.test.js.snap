// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassCard should match the snapshot 1`] = `
<article
  className="class-card"
>
  <header
    className="class-card-header"
  >
    <h3
      className="class-card-name"
    >
      5th Grade Band
    </h3>
  </header>
  <p
    className="class-card-key"
  >
    keeeeeey
  </p>
  <button
    className="class-select-button"
    onClick={[Function]}
  >
    select
  </button>
</article>
`;
